### PR TITLE
Feature/code demo

### LINKED ISSUE #66: Interactive Shell
Use Binder to create an interactive shell on the homepage: ThebeLab + MyBinder: https://thebelab.readthedocs.io/en/latest/

### PR DESCRIPTION
**Pull request by joelachance**
Resolves #66

### CODE REVIEW

**Comment by stefanv** [comment]
@joelachance This is great!  Should we consider a more minimalistic environment with only NumPy installed, and perhaps initialize the terminal with numpy already imported as np?

**Comment by rgommers** [comment]
I like this as well! It's quite responsive, much faster than I'm used to with Binder. I guess it's a function of the set of requirements.

> Should we consider a more minimalistic environment with only NumPy installed, and perhaps initialize the terminal with numpy already imported as np?

Is this for loading speed, or for saving Binder resources, or ... ?

**Comment by stefanv** [comment]
> Is this for loading speed, or for saving Binder resources, or ... ?

It probably won't affect the loading speed too much; more so we have control over the environment we operate in (numpy version, etc.).

If we want to increase response time, we do have the option of hosting (or sponsoring) a MyBinder instance that serves the NumPy image exclusively (most of the start-up time on the MyBinder time is spent moving images to execution nodes, if I understand correctly).

There are further optimizations on the table, such as holding a cache of ready kernels and dishing them out as needed, but that is clearly out of scope for this project.

**Comment by rgommers** [comment]
> It probably won't affect the loading speed too much; more so we have control over the environment we operate in (numpy version, etc.).

Adding just `matplotlib` seems nice, that way we can use nicer demo code. That should not affect the numpy version that gets installed.

And perhaps we can put the `%matplotlib inline` into a config file to always enable it?

**Comment by rgommers** [comment]
@scopatz this is the PR with ThebeLab that we talked about. @joelachance, @scopatz is working on something similar, more general / not Jupyter based (he can explain better). He may be interested to chat/collaborate.

**Comment by scopatz** [comment]
Thanks @rgommers!

@joelachance - can we maybe jump on a call next week sometime? Feel free to email me at scopatz(AT)quansight.com

**Comment by joelachance** [comment]
> This is great! Should we consider a more minimalistic environment with only NumPy installed, and perhaps initialize the terminal with numpy already imported as np?

What we currently have in this PR is the default ThebeLab/Binder image. We can create our own if we'd like, but we will have to host it somewhere. The default image is already very minimal, but as it does take a brief moment to initialize, a smaller image might improve loading time (probably not a huge priority).

A couple of enhancements I think would be nice are as follows:
* A spinner/loader inside of the 'Run' button (in place of the 'Run' text) when the kernel is initializing
* A default numpy import at the top, maybe even a static one above the cursor. I considered removing imports from the default example (but including them when the code is run), but I'm not sure this is best. I think it's good for beginners to know you have to import the library. @rgommers, I agree about including `%matplotlib inline` with the code that is run, but not in the terminal window.

There's one or two more, but I can't remember right now :) I'm on the fence about including `matplotlib`, mostly because of css/styling reasons (the default plot works, but the 'terminal window' isn't going to resize for a larger plot, for example). I think `seaborn` might be included too, but I haven't checked.

**Comment by joelachance** [comment]
@scopatz, I will email you!

File: layouts/partials/css.html [hunk 1]
```diff
@@ -21,5 +21,7 @@
 <!-- Google Fonts - Ubuntu -->
 <link href="https://fonts.googleapis.com/css?family=Ubuntu&display=swap" rel="stylesheet">
 <link href="https://fonts.googleapis.com/css?family=Rubik&display=swap" rel="stylesheet">
```
Comment (Shekharrajak) [inline layouts/partials/css.html]: If you are using Lato font now then can we remove Rubik font link ?

Comment (joelachance) [reply layouts/partials/css.html]: Yes, we can.

I've been playing around with fonts for the main hero title. I'm not sure Ubuntu is the best choice here, although I do think it looks quite nice for body text.

I'll write more soon in #48.

**Comment by rgommers** [comment]
This is in good shape so I'm merging it to not have to migrate it to `master` later. The left over review comments can be discussed/addressed here of course (and fixed with a PR to `master` :) )

**Comment by rgommers** [comment]
Thanks @joelachance and all reviewers, this is really nice!

**Comment by InessaPawson** [comment]
@joelachance I'm submitting the screenshots of the shell to follow up on our last meeting discussion. I'm using Mac Pro. It's the same scenario with Safari and Firefox. I'm unable to get past the third screenshot. It was not an issue while I was in the US.

numpyshell1
numpyshell2
numpyshell3

**Comment by joelachance** [comment]
@InessaPawson The third screenshot is the shell itself-- I had restyled it, but you're the second person to note some confusion about the new styles, so I'm planning on updating this to look more like it did before. If you click at the end of the `import numpy as np` line, you should be able to type more and run your code.

**Comment by InessaPawson** [comment]
A-ha! I guess, my confusion is exacerbated by the fact that I remember the previous version.:) Perhaps, we could  offer a prompt in a darker shade of grey, e.g. “Enter your code here…”
Or something similar to [repl.it](https://repl.it/repls/StimulatingMediumpurpleDisassembly) : “Not sure what to do? Run some examples/an example. Start typing to dismiss.” 
P.S. You have to appreciate the randomly generated name of this repo.;)

**Pull request merged**